### DISCUSSION TITLE
How does funding a payable function in hardhat work?

### DISCUSSION THREAD

**Discussion by cipher222** [smartcontractkit/full-blockchain-solidity-course-py#1545] [Q&A] [answered]
I am at the testing section of the 7th lesson at 11:22, and we just funded a contract using:
 
await fundMe.fund({ value: sendValue }) 
I was a bit confused as to how this works? I did some research on the web and found that you can also add different parameters when calling a payable function such as:

{ from: ethAddress, value: this.web3Loom.utils.toWei(amount) })

Is this utilizing a hardhat package or is Solidity smart enough to know that we are funding it with money? Basically, I want to know how this is working under the hood.

Thanks! 😊

**Comment by cromewar** [comment, marked as answer]
Hello @cipher222, I strongly suggest to move this to the JS course version.

**Reply by cipher222** [reply]
my bad, I accidentally bookmarked the wrong repository. Sorry
